src/zk/correct_key_ni.rs

```diff
@@ -85,7 +85,7 @@ impl NICorrectKeyProof {
                 mask_generation(key_length, &seed_bn) % &ek.n
             })
             .collect::<Vec<BigInt>>();
-        let alpha_primorial: BigInt = BigInt::from_str_radix(&P, 10).unwrap();
+        let alpha_primorial: BigInt = BigInt::from_str_radix(P, 10).unwrap();
```

Comment:
Lovely.  We're doing a big string-to-int conversion in every call to `verify`.  Wow this library has issues. 🙄

Reply:
:(